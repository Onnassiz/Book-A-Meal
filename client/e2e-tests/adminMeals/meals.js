/* eslint func-names: 0, no-unused-expressions: 0 */
const path = require('path');

const imagePath = path.join(__dirname, '../../assets/images/3.jpg');

const meals = {
  'Should sign user into the application': (client) => {
    client
      .url('http://localhost:3002/#/')
      .waitForElementVisible('header')
      .moveToElement('#login', 10, 10)
      .click('#login')
      .waitForElementVisible('h3')
      .assert.containsText('h3', 'Sign In')
      .setValue('#signInEmail', 'onnassiz@gmail.com')
      .setValue('#signInPassword', 'password')
      .click('input[type=submit]')
      .waitForElementVisible('header')
      .assert.visible('#caterer_dropdown')
      .assert.visible('#user_dropdown');
  },

  'Should navigate to the admin meals page': (client) => {
    client
      .moveToElement('#caterer_dropdown', 10, 10)
      .pause(1000)
      .waitForElementVisible('#admin_meals')
      .assert.visible('#admin_meals')
      .click('#admin_meals')
      .waitForElementVisible('#addMeal')
      .pause(2000)
      .assert.visible('#addMeal');
  },

  'Should show new meal modal form': (client) => {
    client
      .moveToElement('#addMeal', 10, 10)
      .pause(1000)
      .assert.containsText('#addMeal', 'Add Meal')
      .click('#addMeal')
      .pause(1000)
      .waitForElementVisible('.box')
      .assert.visible('.box')
      .pause(2000);
  },

  'Should fill meal form and create a new meal': (client) => {
    client
      .pause(1000)
      .setValue('#name', 'Fried Rice')
      .setValue('#price', 2000)
      .setValue('#category', 'Dinner')
      .setValue('#description', 'This is a good meal. It has all you need. Try it out.')
      .moveToElement('input[type=submit]', 10, 10)
      .pause(2000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show a new meal modal form again': (client) => {
    client
      .moveToElement('#addMeal', 10, 10)
      .assert.containsText('#addMeal', 'Add Meal')
      .click('#addMeal')
      .waitForElementVisible('.box')
      .assert.visible('.box')
      .pause(1000);
  },

  'Should fill another meal form and create a new meal': (client) => {
    client
      .pause(1000)
      .setValue('#name', 'Chicken')
      .setValue('#price', 2000)
      .setValue('#category', 'Dinner')
      .setValue('#description', 'This is a good meal. It has all you need. Try it out.')
      .moveToElement('input[type=submit]', 10, 10)
      .pause(2000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show a new meal modal form for the third time': (client) => {
    client
      .moveToElement('#addMeal', 10, 10)
      .assert.containsText('#addMeal', 'Add Meal')
      .click('#addMeal')
      .waitForElementVisible('.box')
      .assert.visible('.box')
      .pause(1000);
  },

  'Should fill the third meal form and create a new meal': (client) => {
    client
      .pause(1000)
      .setValue('#name', 'Beans')
      .setValue('#price', 1000)
      .setValue('#category', 'Dinner')
      .setValue('#description', 'This is a good meal for me and you.')
      .moveToElement('input[type=submit]', 10, 10)
      .pause(2000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show an update meal modal': (client) => {
    client
      .waitForElementVisible('#updateModal:first-of-type')
      .assert.visible('#updateModal:first-of-type')
      .click('#updateModal:first-of-type')
      .assert.visible('.box')
      .pause(1000);
  },

  'Should update the third meal form and update a new meal': (client) => {
    client
      .pause(1000)
      .clearValue('#name')
      .setValue('#name', 'Rice')
      .clearValue('#price')
      .setValue('#price', 1200)
      .moveToElement('input[type=submit]', 10, 10)
      .pause(1000)
      .click('input[type=submit]')
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 5000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default');
  },

  'Should show delete modal and delete meal on button click': (client) => {
    client
      .assert.visible('#deleteModal:first-of-type')
      .click('#deleteModal:first-of-type')
      .waitForElementVisible('button.swal-button.swal-button--confirm.swal-button--danger', 10000)
      .pause(1000)
      .click('button.swal-button.swal-button--confirm.swal-button--danger')
      .waitForElementVisible('button.swal-button.swal-button--confirm', 15000)
      .pause(1000)
      .click('button.swal-button.swal-button--confirm')
      .waitForElementNotVisible('button.swal-button.swal-button--confirm', 20000)
      .assert.hidden('button.swal-button.swal-button--confirm')
      .pause(1000);
  },

  'Should show an image upload modal modal': (client) => {
    client
      .assert.visible('#addPhotoModal:first-of-type')
      .click('#addPhotoModal:first-of-type')
      .waitForElementVisible('#dropzoneHeader')
      .assert.containsText('#dropzoneHeader', 'Drop Image Here')
      .pause(1000)
      .moveToElement('#upload_image', 10, 10)
      .pause(1000)
      .setValue('input[type=file]', imagePath)
      .waitForElementVisible('div.Toastify__toast.Toastify__toast--default', 15000)
      .assert.visible('div.Toastify__toast.Toastify__toast--default')
      .pause(1000);
    client.end();
  },
};

export default meals;
