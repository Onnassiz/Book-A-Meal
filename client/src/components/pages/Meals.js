import React from 'react';
import MealsContainer from '../container/MealsContainer';
import NavBarContainer from '../container/NavBarContainer';

const Meals = () => (
  <div>
    <NavBarContainer page="caterer" />
    <MealsContainer />
  </div>
);

export default Meals;
