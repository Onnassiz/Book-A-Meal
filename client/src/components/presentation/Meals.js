import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import empty from 'is-empty';
import { addProfileModalStyle, deleteModalStyle, addProfileImageModalView } from './../../utilities/modalStyles';
import { BasicInput, TextArea } from './form/BasicInput';
import SubmitButton from '../presentation/form/SubmitButton';
import validateMeal from '../../utilities/validateMealForm';
import Alert from '../presentation/partials/Alert';
import ImageUploader from '../presentation/partials/ImageUploader';
import MealsTableRow from './partials/MealsTableRow';
import loaderImage from '../../../assets/images/loader.gif';

class Meals extends Component {
	constructor(props) {
		super(props);
		this.state = {
			id: '',
			name: '',
			price: '',
			category: '',
			description: '',
			updateMode: false,
			currentMeal: {},
			isShowingModal: false,
			isShowingAddPhoto: false,
			isShowingDeleteMeal: false,
			isShowingDropZoneModal: false,
			errors: {},
		};
		this.handleAddButtonClick = this.handleAddButtonClick.bind(this);
		this.toggleDropZone = this.toggleDropZone.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.onChange = this.onChange.bind(this);
		this.putImage = this.putImage.bind(this);
		this.pushToProfile = this.pushToProfile.bind(this);
		this.toggleUpdateModal = this.toggleUpdateModal.bind(this);
		this.toggleShowDeleteModal = this.toggleShowDeleteModal.bind(this);
		this.toggleAddPhoto = this.toggleAddPhoto.bind(this);
		this.deleteMeal = this.deleteMeal.bind(this);
	}

	componentWillMount() {
		const { user, history, getProfile, profile, meals, getMeals } = this.props;
		if (user.role !== 'caterer') {
			history.push('/');
		}

		if (empty(profile)) {
			getProfile();
		}

		if (empty(meals.meals)) {
			getMeals();
		}
	}

	onChange(e) {
		this.setState({ [e.target.name]: e.target.value });
	}

	toggleDropZone() {
		this.setState({ isShowingDropZoneModal: !this.state.isShowingDropZoneModal });
	}

	putImage(image) {
		const { putImage } = this.props;
		const field = {
			imageUrl: image,
		};

		putImage(this.state.currentMeal.id, field).then((response) => {
			if (response.status === 200) {
				this.toggleAddPhoto({});
			}
		});
	}

	isValid() {
		this.setState({ errors: {} });
		const { errors, isValid } = validateMeal(this.state);
		if (!isValid) {
			this.setState({ errors });
		}
		return isValid;
	}

	handleAddButtonClick() {
		this.resetFields();
		this.setState({ isShowingModal: !this.state.isShowingModal });
	}

	handleSubmit(e) {
		e.preventDefault();
		if (this.isValid()) {
			const { postMeal, updateMeal } = this.props;

			const formData = {
				id: this.state.id,
				name: this.state.name,
				category: this.state.category,
				price: parseInt(this.state.price, 10),
				description: this.state.description,
			};

			if (!this.state.updateMode) {
				postMeal(formData).then((response) => {
					if (response.status === 201) {
						this.resetFields();
						this.handleAddButtonClick();
					}
				});
			} else {
				updateMeal(formData).then((response) => {
					if (response.status === 200) {
						this.resetFields();
						this.handleAddButtonClick();
					}
				});
			}
		}
	}

	pushToProfile() {
		const { history } = this.props;
		history.push('/caterer/business_profile');
	}

	resetFields() {
		this.setState({
			updateMode: false,
			name: '',
			price: '',
			description: '',
			category: '',
			id: '',
		});
	}

	toggleUpdateModal(item) {
		this.setState({
			updateMode: true,
			name: item.name,
			price: item.price.toString(),
			description: item.description === null ? '' : item.description,
			category: item.category,
			id: item.id,
			isShowingModal: !this.state.isShowingModal,
		});
	}

	toggleShowDeleteModal(meal) {
		this.setState({ isShowingDeleteMeal: !this.state.isShowingDeleteMeal, currentMeal: meal });
	}

	toggleAddPhoto(meal) {
		this.setState({ isShowingAddPhoto: !this.state.isShowingAddPhoto, currentMeal: meal });
	}

	deleteMeal() {
		const { deleteMealById } = this.props;
		deleteMealById(this.state.currentMeal.id).then((response) => {
			if (response.status === 200) {
				this.toggleShowDeleteModal(false);
				this.setState({ currentMeal: {} });
			}
		});
	}

	render() {
		const { formState, profile, meals } = this.props;
		const closeModalStyle = {
			float: 'right',
		};

		const SetupProfile = (
			<div className="col-12" style={{ textAlign: 'center' }}>
				<h2>You have not created a Business Profile. You need a business profile before managing meals. Create a business profile first by clicking the link below.</h2>
				<button className="button" onClick={this.pushToProfile}>Setup Profile</button>
			</div>
		);

		return (
			<div id="content2" style={{ paddingBottom: 700 }}>
				{empty(profile.businessName) ? SetupProfile :
					<div className="col-11">
						<button onClick={this.handleAddButtonClick} className="button">Add New Meal</button>
						{empty(meals.alert) ? '' : <Alert alert={meals.alert} />}

						<div>
							<Modal
								isOpen={this.state.isShowingModal}
								closeTimeoutMS={1}
								style={addProfileModalStyle}
								ariaHideApp={false}
								contentLabel="Modal"
							>
								<div className="col-12">
									<a onClick={this.handleAddButtonClick} style={closeModalStyle}><i style={{ fontSize: 25 }} className="material-icons">close</i></a>
								</div>
								<div className="box">
									<h3>{this.state.updateMode ? 'Update Meal' : 'Add Meal'}</h3>
									<div className="show-errors">
										<ul>
											{Object.keys(this.state.errors).map(item => <li key={item}>{ this.state.errors[item] }</li>)}
											{empty(meals.errors) ? '' : Object.keys(meals.errors).map(item => <li key={item}>{ meals.errors[item] }</li>)}
										</ul>
									</div>
									<form onSubmit={this.handleSubmit}>
										<BasicInput name="name" type="text" label="Meal's Name" value={this.state.name} onChange={this.onChange} hasError={this.state.errors.name !== undefined} />
										<BasicInput name="price" type="number" label="Unit Price" value={this.state.price} onChange={this.onChange} hasError={this.state.errors.price !== undefined} />
										<BasicInput name="category" type="text" label="Category" value={this.state.category} onChange={this.onChange} hasError={this.state.errors.category !== undefined} />
										<TextArea name="description" type="text" label="Description" value={this.state.description} onChange={this.onChange} />
										<SubmitButton value={this.state.updateMode ? 'Update' : 'Submit'} isLoading={formState.isLoading} />
									</form>
								</div>
							</Modal>
						</div>

						<div>
							<Modal
								isOpen={this.state.isShowingDeleteMeal}
								closeTimeoutMS={1}
								style={deleteModalStyle}
								ariaHideApp={false}
								contentLabel="Modal"
							>
								<div>
									<p>Meal: <b>{this.state.currentMeal.name}</b></p>
									<p>Are you sure you want to delete this meal?</p>
									<br />
									<p>
										<button onClick={this.toggleShowDeleteModal} className="button-default">Close</button>
										<button onClick={this.deleteMeal} style={{ float: 'right' }} className="button-error" disabled={formState.isLoading}>Delete</button>
										<img style={{ float: 'right' }} src={loaderImage} alt="loader" hidden={!formState.isLoading} />
									</p>
								</div>
							</Modal>
						</div>

						<div>
							<Modal
								isOpen={this.state.isShowingAddPhoto}
								closeTimeoutMS={1}
								style={addProfileImageModalView}
								ariaHideApp={false}
								contentLabel="Modal"
							>
								<div className="col-12">
									<div className="col-12">
										<a onClick={this.toggleAddPhoto} style={{ float: 'right' }}><i style={{ fontSize: 25 }} className="material-icons">close</i></a>
									</div>
									<ImageUploader putImage={this.putImage} />
								</div>
							</Modal>
						</div>

						<div className="col-12">
							{empty(meals.meals) ? '' :
								<div>
									<table>
										<thead>
											<tr>
												<th>S/N</th>
												<th>Meal Name</th>
												<th>Price</th>
												<th>Category</th>
												<th>Date Created</th>
												<th>Added By</th>
												<th>Actions</th>
											</tr>
										</thead>
										<tbody>
											{meals.meals.map((item, i) => <MealsTableRow toggleAddPhoto={() => this.toggleAddPhoto(item)} toggleShowDeleteModal={() => this.toggleShowDeleteModal(item)} toggleUpdateModal={() => this.toggleUpdateModal(item)} item={item} key={item.id} i={i + 1} />)}
										</tbody>
									</table>
								</div>
							}
						</div>
					</div>
				}
			</div>
		);
	}
}

Meals.propTypes = {
	user: PropTypes.object.isRequired,
	meals: PropTypes.object.isRequired,
	history: PropTypes.object.isRequired,
	profile: PropTypes.object.isRequired,
	formState: PropTypes.object.isRequired,
	postMeal: PropTypes.func.isRequired,
	updateMeal: PropTypes.func.isRequired,
	deleteMealById: PropTypes.func.isRequired,
	getMeals: PropTypes.func.isRequired,
	putImage: PropTypes.func.isRequired,
	getProfile: PropTypes.func.isRequired,
};

export default Meals;
