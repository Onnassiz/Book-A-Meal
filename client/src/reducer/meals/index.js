import { SET_MEAL, SET_MEAL_SERVER_ERRORS } from './actions';

const initialState = {
  meals: [],
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_MEAL:
      return { ...state, meals: action.meals };
    case SET_MEAL_SERVER_ERRORS:
      return { ...state, errors: action.errors };
    default:
      return state;
  }
};
