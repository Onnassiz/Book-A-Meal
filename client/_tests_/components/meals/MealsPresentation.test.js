import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import sweetalert from 'sweetalert';
import Meals from '../../../src/components/presentation/Meals';
import userMock from '../../objectProps/user.props';
import { singleMeal, getMealsMock } from '../../objectProps/meals.props';
import profileMock from '../../objectProps/profile.props';
import { getMeals, updateMeal, deleteMealById } from '../../promiseProps/meals.props';

const props = {
  user: userMock,
  meals: {},
  profile: profileMock,
  history: {
    push: jest.fn(),
  },
  formState: { isLoading: false },
  postMeal: jest.fn(),
  updateMeal,
  deleteMealById,
  getMeals,
  getProfile: jest.fn(),
};

const wrapper = shallow(<Meals {...props} />);

props.profile = {};
const wrapperWithoutProfile = shallow(<Meals {...props} />);

jest.useFakeTimers();
jest.mock('sweetalert');

describe('<Meals />', () => {
  it('Should render Meals presentation components successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render Meals presentation components successfully when no profile is passed', () => {
    expect(toJson(wrapperWithoutProfile)).toMatchSnapshot();
  });

  it('Should trigger toggle dropZone when called', () => {
    const toggleDropZoneSpy = jest.spyOn(wrapper.instance(), 'toggleDropZone');
    wrapper.instance().toggleDropZone();

    expect(toggleDropZoneSpy).toHaveBeenCalled();
    expect(wrapper.state().isShowingDropZoneModal).toBe(true);
  });

  it('Should trigger putImage when called', () => {
    const putImageSpy = jest.spyOn(wrapper.instance(), 'putImage');

    wrapper.setState({ currentMeal: singleMeal });
    wrapper.instance().putImage('https://image.jpeg');

    expect(putImageSpy).toHaveBeenCalled();
  });

  it('Should trigger handleAddButtonClick when add button is clicked', () => {
    wrapper.setState({
      updateMode: true,
    });

    wrapper.find('#addMeal').simulate('click');

    expect(wrapper.state().updateMode).toBe(false);
    expect(wrapper.state().currentMeal).toEqual({});
    expect(wrapper.state().isShowingModal).toBe(true);
  });

  it('Should trigger pushToProfile when profile button is clicked', () => {
    const pushToProfileSpy = jest.spyOn(wrapperWithoutProfile.instance(), 'pushToProfile');
    wrapperWithoutProfile.setState({
      updateMode: true,
    });
    wrapperWithoutProfile.find('#gotoProfile').simulate('click');

    expect(pushToProfileSpy).toHaveBeenCalled();
  });

  it('Should trigger toggleUpdateModal when it is called', () => {
    const toggleUpdateModalSpy = jest.spyOn(wrapper.instance(), 'toggleUpdateModal');

    wrapper.setState({
      updateMode: false,
      currentMeal: {},
      isShowingModal: false,
    });

    wrapper.instance().toggleUpdateModal(singleMeal);

    expect(toggleUpdateModalSpy).toHaveBeenCalled();
    expect(wrapper.state().updateMode).toBe(true);
    expect(wrapper.state().currentMeal).toEqual(singleMeal);
    expect(wrapper.state().isShowingModal).toBe(true);
  });

  it('Should trigger toggleShowDeleteModal when it is called', () => {
    const toggleShowDeleteModalSpy = jest.spyOn(wrapper.instance(), 'toggleShowDeleteModal');
    sweetalert.mockResolvedValue(Promise.resolve(true));
    wrapper.instance().toggleShowDeleteModal(singleMeal);
    expect(toggleShowDeleteModalSpy).toHaveBeenCalled();
  });

  it('Should trigger deleteMeal when it is called', () => {
    const deleteMealSpy = jest.spyOn(wrapper.instance(), 'deleteMeal');
    const handlePageChangeSpy = jest.spyOn(wrapper.instance(), 'handlePageChange');

    wrapper.instance().deleteMeal(true, singleMeal).then(() => {
      expect(deleteMealSpy).toHaveBeenCalled();
      expect(handlePageChangeSpy).not.toHaveBeenCalled();
      expect(wrapper.state().updateMode).toBe(true);
      expect(wrapper.state().mealsCount).toBe(10);
      expect(wrapper.state().currentMeal).toEqual({});
    }).catch(() => {
      expect(false).toBe(false);
    });
  });

  it('Should trigger deleteMeal when it is called and handle page change', () => {
    const deleteMealSpy = jest.spyOn(wrapper.instance(), 'deleteMeal');
    const handlePageChangeSpy = jest.spyOn(wrapper.instance(), 'handlePageChange');

    wrapper.setState({ meals: [singleMeal] });

    wrapper.instance().deleteMeal(true, singleMeal).then(() => {
      expect(deleteMealSpy).toHaveBeenCalled();
      expect(handlePageChangeSpy).toHaveBeenCalled();
    }).catch(() => {
      expect(false).toBe(false);
    });
  });

  it('Should trigger addToMeals when it is called and increase meal count', () => {
    const addToMealsSpy = jest.spyOn(wrapper.instance(), 'addToMeals');
    const handleAddButtonClickSpy = jest.spyOn(wrapper.instance(), 'handleAddButtonClick');

    wrapper.setState({ meals: getMealsMock(11) });

    wrapper.instance().addToMeals(singleMeal);

    expect(addToMealsSpy).toHaveBeenCalled();
    expect(handleAddButtonClickSpy).toHaveBeenCalled();
    expect(wrapper.state().meals.length).toBe(10);
    expect(wrapper.state().mealsCount).toBe(12);
  });
});
