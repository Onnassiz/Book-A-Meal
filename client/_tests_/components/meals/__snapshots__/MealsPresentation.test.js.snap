// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meals /> Should render Meals presentation components successfully 1`] = `
<div
  id="content-body"
>
  <div
    className="col-12"
  >
    <button
      className="button"
      id="addMeal"
      onClick={[Function]}
    >
      <i
        className="ion-android-add"
      />
       Add Meal
    </button>
    <div>
      <div>
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <a
            activePage={1}
            firstPageText="«"
            getPageUrl={[Function]}
            hideDisabled={true}
            hideFirstLastPages={false}
            innerClass="pagination"
            itemsCountPerPage={10}
            lastPageText="»"
            nextPageText="⟩"
            onChange={[Function]}
            pageRangeDisplayed={5}
            prevPageText="⟨"
            totalItemsCount={11}
          />
        </div>
      </div>
      <div>
        <MealModal
          addToMeals={[Function]}
          formState={
            Object {
              "isLoading": false,
            }
          }
          handleAddButtonClick={[Function]}
          isShowingModal={false}
          meals={Object {}}
          postMeal={[MockFunction]}
          state={
            Object {
              "activePage": 1,
              "currentMeal": Object {},
              "isShowingAddPhoto": false,
              "isShowingDropZoneModal": false,
              "isShowingModal": false,
              "meals": Array [
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 0,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 40,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 80,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 120,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 160,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 200,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 240,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 280,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 320,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 360,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5333",
                  "imageUrl": "http://thisImage.jpeg",
                  "name": "Malvaceae",
                  "price": 400,
                  "userId": "2794fddd-14f3-4ec9-a1dc-88ad2bc649da",
                },
              ],
              "mealsCount": 11,
              "updateMode": false,
            }
          }
          updateComponentMeals={[Function]}
          updateMeal={[Function]}
        />
        <MealsTable
          state={
            Object {
              "activePage": 1,
              "currentMeal": Object {},
              "isShowingAddPhoto": false,
              "isShowingDropZoneModal": false,
              "isShowingModal": false,
              "meals": Array [
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 0,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 40,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 80,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 120,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 160,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 200,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 240,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 280,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 320,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5444",
                  "name": "Malvaceae",
                  "price": 360,
                  "userId": "84c6d80c-3d4a-4772-856c-0b3825bc537e",
                },
                Object {
                  "category": "dolor sit",
                  "description": "Maecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat. Curabitur gravida nisi at nibh.",
                  "id": "84c6d80c-3d4a-4772-856c-0b3825bc5333",
                  "imageUrl": "http://thisImage.jpeg",
                  "name": "Malvaceae",
                  "price": 400,
                  "userId": "2794fddd-14f3-4ec9-a1dc-88ad2bc649da",
                },
              ],
              "mealsCount": 11,
              "updateMode": false,
            }
          }
          toggleAddPhoto={[Function]}
          toggleShowDeleteModal={[Function]}
          toggleUpdateModal={[Function]}
        />
      </div>
      <div>
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <a
            activePage={1}
            firstPageText="«"
            getPageUrl={[Function]}
            hideDisabled={true}
            hideFirstLastPages={false}
            innerClass="pagination"
            itemsCountPerPage={10}
            lastPageText="»"
            nextPageText="⟩"
            onChange={[Function]}
            pageRangeDisplayed={5}
            prevPageText="⟨"
            totalItemsCount={11}
          />
        </div>
      </div>
      <ImageUploader
        isShowingAddPhoto={false}
        putImage={[Function]}
        toggleAddPhoto={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`<Meals /> Should render Meals presentation components successfully when no profile is passed 1`] = `
<div
  id="content-body"
>
  <div
    className="col-12"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h2>
      You have not created a Business Profile. You need a business profile before managing meals. Create a business profile first by clicking the link below.
    </h2>
    <button
      className="button"
      id="gotoProfile"
      onClick={[Function]}
    >
      Setup Profile
    </button>
  </div>
</div>
`;
